pkg/services/searchV2/bluge.go

```diff
@@ -30,145 +30,212 @@ const (
 	documentFieldInternalID  = "__internal_id" // only for migrations! (indexed as a string)
 )
 
-func initBlugeIndex(dashboards []dashboard, logger log.Logger) (*bluge.Reader, error) {
-	config := bluge.InMemoryOnlyConfig()
-
-	// open an index writer using the configuration
-	writer, err := bluge.OpenWriter(config)
+func initIndex(dashboards []dashboard, logger log.Logger) (*bluge.Reader, *bluge.Writer, error) {
+	writer, err := bluge.OpenWriter(bluge.InMemoryOnlyConfig())
 	if err != nil {
-		return nil, fmt.Errorf("error opening writer: %v", err)
+		return nil, nil, fmt.Errorf("error opening writer: %v", err)
 	}
-	defer func() {
-		err = writer.Close()
```

Comment:
spent a lot of time finding out why updates do not work - turned out due to closed writer (we did not use it before after first indexing)